### PR TITLE
[DOJOT-14] Update: Allow register specific identification (ID) for one device

### PR DESCRIPTION
**Pull request by rodrigoaasm**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Is there any issue related to this PR in other repository?** (such as dojot/dojot)
[DOJOT-14]: https://dojot.atlassian.net/browse/DOJOT-14

* **Other information**:

### CODE REVIEW

File: DeviceManager/DeviceHandler.py [hunk 1]
```diff
@@ -386,6 +386,20 @@ def get_device(token, device_id, sensitive_data=False):
...
+        """
+        Validates if the device id follows the rules implemented by dojot 
+
+        :param device_id: The device id
+
+        :raises ValidationError: If the device id violated the rules or is too long
+        """
+        regex = re.compile(r'^[0-9a-fA-F]{2,6}$')
+        if regex.match(device_id) == None:
+            raise ValidationError('Device ID must be 6 characters and must be hexadecimal (0-9,a-f,A-F).')
```
Comment (rascaraficci) [inline DeviceManager/DeviceHandler.py:400]: Device ID must be 2-6 characters ....

File: DeviceManager/DeviceHandler.py [hunk 2]
```diff
@@ -432,7 +446,13 @@ def create_device(cls, params, token):
                 data_request = params.get('data')
                 device_data, json_payload = parse_payload(content_type, data_request, device_schema)
                 validate_repeated_attrs(json_payload)
-                device_data['id'] = DeviceHandler.generate_device_id()
+
+                if json_payload.get('id', None) is None or count > 1 : 
+                    device_data['id'] = DeviceHandler.generate_device_id()
+                else:
+                    DeviceHandler.validate_device_id(json_payload['id'])
```
Comment (rascaraficci) [inline DeviceManager/DeviceHandler.py:453]: Acho que seria importante verificarmos se o id não está em uso para evitar sobrescrever um dispositivo existente.

File: docs/apiary.apib [hunk 3]
```diff
@@ -662,6 +662,107 @@ attributes to be applied to this device.
...
+                    "a device can not have repeated attributes"
+                ],
+                "status": 400
+            }
+
++ Response 400 (application/json)
+
+            {
+                "message": [
+                    "Device ID must be 6 characters and must be hexadecimal (0-9,a-f,A-F)"
```
Comment (rascaraficci) [inline docs/apiary.apib:748]: 2-6 characters

**Review by rascaraficci — COMMENTED** [review]
I left some comments.

File: DeviceManager/DeviceHandler.py (hunk 2 above)
Comment (rodrigoaasm) [reply DeviceManager/DeviceHandler.py:453]: Já existe essa verificação implementada no código. Não foi preciso adicioná-la

**Review by rascaraficci — APPROVED** [review]
It looks good to me!

**Review by luissajunior — APPROVED** [review]

**Pull request merged**